### PR TITLE
immutable map index integration

### PR DESCRIPTION
**Pull request by IvanPleshkov**
### All Submissions:

* [ ] Contributions should target the `dev` branch. Did you create your branch from `dev`?
* [ ] Have you followed the guidelines in our Contributing document?
* [ ] Have you checked to ensure there aren't other open [Pull Requests](../../../pulls) for the same update/change?

### New Feature Submissions:

1. [ ] Does your submission pass tests?
2. [ ] Have you formatted your code locally using `cargo +nightly fmt --all` command prior to submission?
3. [ ] Have you checked your code using `cargo clippy --all --all-features` command?

### Changes to Core Features:

* [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your core changes, as applicable?
* [ ] Have you successfully ran tests with your changes locally?

### CODE REVIEW

File: lib/segment/src/segment_constructor/segment_constructor_base.rs [hunk 1]
```diff
@@ -178,6 +188,7 @@ fn create_segment(
         SegmentType::Plain
     };
     let appendable_flag = vector_data.values().all(VectorData::is_appendable);
+    assert_eq!(is_appendaple_by_config, appendable_flag);
```
Comment (generall) [inline lib/segment/src/segment_constructor/segment_constructor_base.rs:191]:
```
docker run --rm -it --network=host qdrant/bfb:dev ./bfb --uri 'http://localhost:6334' -b 50 --indexing-threshold 0 --mmap-threshold 100
```

causes the assertion failure

File: lib/segment/src/segment.rs [hunk 2]
```diff
@@ -1325,6 +1325,7 @@ impl SegmentEntry for Segment {
     ) -> OperationResult<bool> {
         self.handle_version_and_failure(op_num, None, |segment| match field_type {
             Some(schema) => {
+                // todo(ivan)
```
Comment (generall) [inline lib/segment/src/segment.rs:1328]: todo what?

File: lib/segment/src/segment_constructor/segment_constructor_base.rs [hunk 3]
```diff
@@ -81,11 +81,21 @@ fn create_segment(
 
     let id_tracker = sp(SimpleIdTracker::open(database.clone())?);
 
+    let is_appendaple_by_config = config
```
Comment (generall) [inline lib/segment/src/segment_constructor/segment_constructor_base.rs:84]:
```suggestion
    let is_appendable_by_config = config
```

File: lib/segment/src/segment_constructor/segment_constructor_base.rs (hunk 1 above)
Comment (IvanPleshkov) [reply lib/segment/src/segment_constructor/segment_constructor_base.rs]: Extremely good catch, thanks for it!

Comment (IvanPleshkov) [reply lib/segment/src/segment_constructor/segment_constructor_base.rs]: fixed

File: lib/segment/src/segment.rs (hunk 2 above)
Comment (IvanPleshkov) [reply lib/segment/src/segment.rs]: Sorry, it was obsolete todo that I already done, removed

File: lib/segment/src/segment_constructor/segment_constructor_base.rs (hunk 3 above)
Comment (IvanPleshkov) [reply lib/segment/src/segment_constructor/segment_constructor_base.rs]: done

**Review by generall — APPROVED** [review]

**Pull request merged**